### ISSUE TITLE
Files with all rules ignored are still parsed.

### ISSUE THREAD

**Issue by brackendawson** [adrienverge/yamllint#539]
If you ignore a file because it's very large and you don't want to spend the time linting it, you don't save any time.

# Steps to re-create
1. Make a large (like 2MB) and small yaml file and this configuration:
   ```
   small.yaml
   large.yaml
   .lint-config.yaml
   ```
   ```yaml
   extends: default
   
   ignore: |
     large.yaml
   ```
2. Run `time yamllint -c .lint-config.yaml .`
3. `rm large.yaml`
4. Run `time yamllint -c .lint-config.yaml .` again.

# Expected results
Step 2 and 4 take the same time.

# Actual results
Step 4 is a lot faster than step 2.

**Comment by andrewimeson** [comment]
Can't reproduce with yamllint v1.29.0

```sh-session
$ base64 /dev/urandom | head -c 10000000 > big.yaml
$ ls -lah
total 19552
drwxr-xr-x   5 andrewimeson  staff   160B Feb 15 18:18 .
drwxr-xr-x  24 andrewimeson  staff   768B Feb 15 18:09 ..
-rw-r--r--   1 andrewimeson  staff    44B Feb 15 18:18 .lint-config.yaml
-rw-r--r--   1 andrewimeson  staff   9.5M Feb 15 18:13 big.yaml
-rw-r--r--   1 andrewimeson  staff    41B Feb 15 18:14 small.yaml

$ time yamllint .
./small.yaml
  3:23      warning  truthy value should be one of [false, true]  (truthy)

./big.yaml
  1:1       warning  missing document start "---"  (document-start)
  1:10000001error    no new line character at the end of file  (new-line-at-end-of-file)

yamllint .  7.48s user 0.06s system 99% cpu 7.581 total

$ time yamllint -c .lint-config.yaml .
./small.yaml
  3:23      warning  truthy value should be one of [false, true]  (truthy)

yamllint -c .lint-config.yaml .  0.06s user 0.02s system 64% cpu 0.123 total
```

What version of yamllint are you running?

**Comment by adrienverge** [comment]
Hello, same here with yamllint 1.29.0:
```bash
$ time yamllint -d '{extends: default}' .
7,69s user 0,06s system 99% cpu 7,773 total
```
```bash
$ time yamllint -d '{extends: default, ignore: big.yaml}' .
0,11s user 0,02s system 98% cpu 0,139 total
```

**Comment by brackendawson** [comment]
> What version of yamllint are you running?

1.29.0

macOS 13.1 on a Macbook Pro 16-inch 2021 (M1)

**Comment by brackendawson** [comment]
I can't recreate it either this morning, I must have spelled "large.yaml" wrong or something.
